pkg/files/files.go

```diff
@@ -92,6 +98,24 @@ func (h *Handler) writeKubeConfig(res *engine.Result) error {
 	return nil
 }
 
+func (h *Handler) writePrivateKey(res *engine.Result) error {
+	baseDir, err := GetBaseDir()
+	if err != nil {
+		return err
+	}
+	if err := h.fs.MkdirAll(baseDir, 0755); err != nil {
+		return err
+	}
+	if err := ioutil.WriteFile(
+		filepath.Join(baseDir, privateKeyFile),
+		[]byte(res.PrivateKeyContent),
+		0644); err != nil {
+		return err
+	}
```

Comment:
Just FYI. This is the reason why I do not like this style personally. I have to read very carefully and invest more energy than usually to understand what this is all about. Having actual business logic invocation and error handling separated makes always more sense to me because it is easier readable for me personally.

Reply:
Ack and agree, it's clear here what you mean, I've pushed changes to move the checks on each of this long function calls to a separate statement.